etl/src/main/java/org/pih/hivmigration/etl/sql/TbStatusMigrator.groovy

```diff
@@ -0,0 +1,223 @@
+            ;
+        ''')
+
+        // Populate row for any patients who have a directly referenced intake encounter
+
+        executeMysql("Load status rows that are directly linked to intake encounters", '''
+            update      hivmigration_tb_status_intake i
+            inner join  hivmigration_tb_status s on i.source_encounter_id = s.source_encounter_id
+            set         i.STATUS_ON_ENCOUNTER = true,
+                        i.STATUS_DATE = s.STATUS_DATE, 
+                        i.ENTERED_DATE = s.ENTERED_DATE, 
+                        i.ENTERED_BY = s.ENTERED_BY,
+                        i.PPD_POSITIVE_P = s.PPD_POSITIVE_P, 
+                        i.TB_ACTIVE_P = s.TB_ACTIVE_P, 
+                        i.EXTRAPULMONARY_P = s.EXTRAPULMONARY_P, 
+                        i.PULMONARY_P = s.PULMONARY_P, 
+                        i.DRUG_RESISTANT_P = s.DRUG_RESISTANT_P, 
+                        i.DRUG_RESISTANT_COMMENT = s.DRUG_RESISTANT_COMMENT
+        ''')
+
+        // For the remaining patients, iterate over statuses that are not linked to an encounter, and use to populate
+
+        executeMysql("Add aggregate information to intake where multiple patient-level rows exist", '''
+            update        hivmigration_tb_status_intake i
+            inner join    (
+                            select    s.source_patient_id,
+                                      min(s.status_date) as first_status_date,
+                                      sum(if(s.ppd_positive_p = 't', 1, 0)) as num_ppd_positive,
+                                      sum(if(s.tb_active_p = 't', 1, 0)) as num_tb_active,
+                                      sum(if(s.extrapulmonary_p = 't', 1, 0)) as num_extrapulmonary,
+                                      sum(if(s.pulmonary_p = 't', 1, 0)) as num_pulmonary,
+                                      sum(if(s.drug_resistant_p = 't', 1, 0)) as num_drug_resistant,
+                                      group_concat(s.drug_resistant_comment, ' , ') as all_drug_resistant_comment
+                            from      hivmigration_tb_status s
+                            where     s.source_encounter_id is null
+                            group by  s.source_patient_id
+                          ) s on i.source_patient_id = s.source_patient_id
+            set           i.STATUS_DATE = s.first_status_date, 
+                          i.PPD_POSITIVE_P = if(num_ppd_positive > 0, 't', 'f'), 
+                          i.TB_ACTIVE_P = if(num_tb_active > 0, 't', 'f'), 
+                          i.EXTRAPULMONARY_P = if(num_extrapulmonary > 0, 't', 'f'), 
+                          i.PULMONARY_P = if(num_pulmonary > 0, 't', 'f'), 
+                          i.DRUG_RESISTANT_P = if(num_drug_resistant > 0, 't', 'f'), 
+                          i.DRUG_RESISTANT_COMMENT = s.all_drug_resistant_comment
+            where         i.status_on_encounter = false
+            ;
```

Comment:
Is the assumption here that all statuses not linked to an encounter came from an Intake, not a follow-up?  There are about 500,000+ rows in the TB Status table linked to follow-up encounters.

Reply:
No.  It doesn't make that assumption.  It's just trying to establish if the unlinked data could presumably be migrated in for the intake encounter.

Reply:
Maybe I'm not following, but it seems to be making that assumption... or, I guess to be clearer, it's saying "if we don't have a linked Intake encounter for a patient, assume that any row not linked to an encounter can be used to populate the Intake in OpenMRS", because this query is just pulling all the rows in TB status without an encounter and then linking them to to the hivmigration_tb_status_intake for all rows/patients that haven't been populated by a linked intake encounter?

Reply:
Yes, I think you are right.  We can debate this.  It's possible we want to change this, but i'm not entirely sure the best way to do so.

Reply:
It's possible that that assumption may be fine... do we know the history of statuses that aren't linked to encounters?  

We also could skip this part entirely has legacy data from v1/v2/not MVP if we know that all the v3 Intakes are linked to statuses by encounter_id, so we have those covered.